### PR TITLE
SDIT-1421 Add association to offender roles to questionnaire and general tidy up

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireService.kt [hunk 1]
```diff
@@ -63,10 +64,14 @@ class QuestionnaireService(
           dateRequired = questionnaireAnswer.dateRequired,
         )
           .apply {
-            if (includeNextQuestion) {
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireService.kt:66]: Had a look at the full response in preprod and there was just too much unnecessary information - so just pulled in minimum for identification an readability

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt [hunk 2]
```diff
@@ -36,15 +36,23 @@ data class Incident(
...
-  val requirements: MutableList<IncidentRequirement> = mutableListOf(),
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "QUESTIONNAIRE_ID", nullable = false)
+  val questionnaire: Questionnaire,
 
-  @OneToMany(mappedBy = "id.incident", cascade = [CascadeType.ALL], orphanRemoval = true)
+  @OneToMany(mappedBy = "id.incidentId", cascade = [CascadeType.ALL], orphanRemoval = true)
   val questions: MutableList<IncidentQuestion> = mutableListOf(),
 
+  @OneToMany(mappedBy = "id.incidentId", cascade = [CascadeType.ALL], orphanRemoval = true)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/Incident.kt:50]: Another PR is coming shortly to separate offender and staff parties

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/IncidentParty.kt [hunk 3]
```diff
@@ -32,11 +32,9 @@ class IncidentParty(
   @EmbeddedId
   val id: IncidentPartyId,
 
-  @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/IncidentParty.kt:35]: don't need incident here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/IncidentRequirement.kt [hunk 4]
```diff
@@ -30,10 +30,6 @@ class IncidentRequirement(
   @EmbeddedId
   val id: IncidentRequirementId,
 
-  @ManyToOne(fetch = FetchType.LAZY, optional = false)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/IncidentRequirement.kt:33]: don't need incident here either

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResourceIntTest.kt [hunk 5]
```diff
@@ -43,12 +43,20 @@ class QuestionnaireResourceIntTest : IntegrationTestBase() {
           questionnaireAnswer(answer = "Q1A1: Yes", listSequence = 1, answerSequence = 1, nextQuestion = question2)
           questionnaireAnswer(answer = "Q1A2: No", listSequence = 2, answerSequence = 2, nextQuestion = question2)
         }
+        offenderRole("ABS")
+        offenderRole("ESC")
+        offenderRole("FIGHT")
+        offenderRole("PERP")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResourceIntTest.kt:49]: I wonder if you could create a version of this that accepts a collection, so you can write `offenderRoles("ABS","ESC","FIGHT","PERP"). Might reduce a little noise.

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResourceIntTest.kt (hunk 5 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResourceIntTest.kt:49]: will try and tidy it up

**Pull request merged**